src/main/java/org/opentripplanner/graph_builder/module/osm/naming/SidewalkNamer.java

```diff
@@ -0,0 +1,290 @@
+  }
+
+  @Override
+  public void recordEdges(OSMWithTags way, StreetEdgePair pair) {
+    // this way is a sidewalk and hasn't been named yet (and is not explicitly unnamed)
+    if (way.isSidewalk() && way.hasNoName() && !way.isExplicitlyUnnamed()) {
+      pair
+        .asIterable()
+        .forEach(edge -> unnamedSidewalks.add(new EdgeOnLevel(edge, way.getLevels())));
+    }
+    // the way is _not_ a sidewalk and does have a name
+    else if (way.isNamed() && !way.isLink()) {
+      // We generate two edges for each osm way: one there and one back. This spatial index only
+      // needs to contain one item for each road segment with a unique geometry and name, so we
+      // add only one of the two edges.
+      var edge = pair.pickAny();
+      streetEdges.insert(
+        edge.getGeometry().getEnvelopeInternal(),
+        new EdgeOnLevel(edge, way.getLevels())
+      );
+    }
+  }
+
+  @Override
+  public void postprocess() {
+    ProgressTracker progress = ProgressTracker.track(
+      "Assigning names to sidewalks",
+      500,
+      unnamedSidewalks.size()
+    );
+
+    this.preciseBuffer = new PreciseBuffer(computeEnvelopeCenter(), BUFFER_METERS);
+
+    final AtomicInteger namesApplied = new AtomicInteger(0);
+    unnamedSidewalks
+      .parallelStream()
+      .forEach(sidewalkOnLevel -> {
+        assignNameToSidewalk(sidewalkOnLevel, namesApplied);
+
+        // Keep lambda! A method-ref would cause incorrect class and line number to be logged
+        // noinspection Convert2MethodRef
+        progress.step(m -> LOG.info(m));
+      });
+
+    LOG.info(
+      "Assigned names to {} of {} of sidewalks ({}%)",
+      namesApplied.get(),
+      unnamedSidewalks.size(),
+      DoubleUtils.roundTo2Decimals((double) namesApplied.get() / unnamedSidewalks.size() * 100)
+    );
+
+    LOG.info(progress.completeMessage());
+
+    // set the indices to null so they can be garbage-collected
+    streetEdges = null;
+    unnamedSidewalks = null;
+  }
+
+  /**
+   * Compute the centroid of all sidewalk edges.
+   */
+  private Coordinate computeEnvelopeCenter() {
+    var envelope = new Envelope();
+    unnamedSidewalks.forEach(e -> {
+      envelope.expandToInclude(e.edge.getFromVertex().getCoordinate());
+      envelope.expandToInclude(e.edge.getToVertex().getCoordinate());
+    });
+    return envelope.centre();
+  }
+
+  /**
+   * The actual worker method that runs the business logic on an individual sidewalk edge.
+   */
+  private void assignNameToSidewalk(EdgeOnLevel sidewalkOnLevel, AtomicInteger namesApplied) {
+    var sidewalk = sidewalkOnLevel.edge;
+    var buffer = preciseBuffer.preciseBuffer(sidewalk.getGeometry());
+    var sidewalkLength = SphericalDistanceLibrary.length(sidewalk.getGeometry());
+
+    var candidates = streetEdges.query(buffer.getEnvelopeInternal());
+
+    groupEdgesByName(candidates)
+      // make sure we only compare sidewalks and streets that are on the same level
```

Comment:
Capitalize first letter of sentence.

Reply:
https://github.com/opentripplanner/OpenTripPlanner/pull/5774/commits/60c4ee925f51689bacd596f517ea709984b36615